### PR TITLE
Task6

### PR DESCRIPTION
**Pull request by dheerajthodupunoori**

### CODE REVIEW

File: Week6/news/src/app/auth-guard.service.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { Router, CanActivate } from "@angular/router";
+import { AuthService } from "./auth.service";
+@Injectable({
+  providedIn: "root"
+})
+export class AuthGuardService implements CanActivate {
+  constructor(private _authService: AuthService, private router: Router) {}
+  canActivate(): boolean {
+    if (!this._authService.isUserLoggedIn()) {
+      this.router.navigate([""]);
```
Comment (aditya365) [inline Week6/news/src/app/auth-guard.service.ts]: navigate to login page

File: Week6/news/src/app/auth/auth.component.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    private _authService: AuthService
+  ) {}
+
+  ngOnInit() {
+    this.isLoggedIn = Boolean(localStorage.getItem("isLoggedIn"));
+  }
+
+  showLogInForm() {
+    this.displayLogInForm = !this.displayLogInForm;
+    console.log(this.displayLogInForm);
```
Comment (aditya365) [inline Week6/news/src/app/auth/auth.component.ts]: remove console logs

File: Week6/news/src/app/app.component.html [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<div>
+  <app-auth></app-auth>
```
Comment (aditya365) [inline Week6/news/src/app/app.component.html]: why are you including auth component here? Shouldn't it be in a separate route?

File: Week6/news/src/app/news.service.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import { Observable } from "rxjs";
+
+@Injectable({
+  providedIn: "root"
+})
+export class NewsService {
+  public sourceName = "bbc-news";
+  public articles = [];
+  constructor(private http: HttpClient) {
+    this.getAuthorsList(this.sourceName).then(data => (this.articles = data));
```
Comment (aditya365) [inline Week6/news/src/app/news.service.ts]: constructor should not contain complex/blocking operations. You just have to call the getAuthorsList method from component in onInit.

File: Week6/news/src/app/news.service.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  getChannels(): Observable<any> {
+    return this.http.get("https://newsapi.org/v1/sources");
+  }
+
+  getAuthorsList(channelId) {
+    let channelAuthorEndpoint = newsApiEndPoint.articlesEndPoint;
+    const url = `${channelAuthorEndpoint}source=${channelId}&apiKey=${newsApiEndPoint.key}`;
+    return fetch(url)
```
Comment (aditya365) [inline Week6/news/src/app/news.service.ts]: why fetch here instead of httpClient?

File: Week6/news/src/app/app.component.html (hunk 3 above)
Comment (dheerajthodupunoori) [reply Week6/news/src/app/app.component.html]: The naming convention was wrong , actually its a header

File: Week6/news/src/app/news.service.ts (hunk 5 above)
Comment (dheerajthodupunoori) [reply Week6/news/src/app/news.service.ts]: no specific reason , i already used httpclient for another method .  So just copied code from previous task .

File: Week6/news/src/app/auth-guard.service.ts (hunk 1 above)
Comment (dheerajthodupunoori) [reply Week6/news/src/app/auth-guard.service.ts]: The route("") itself contains login form , which was hidden intially . No seperate component for login page .

**Pull request merged**